sw/host/opentitanlib/src/transport/ti50emulator/uart.rs

```diff
@@ -2,41 +2,118 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
-use anyhow::Result;
+use anyhow::{bail, Context, Result};
+
+use std::cell::{Cell, RefCell};
+use std::io::{Read, Write};
+use std::os::unix::net::UnixStream;
+use std::path::PathBuf;
+use std::rc::Rc;
 use std::time::Duration;
 
-use crate::io::uart::Uart;
-use crate::transport::TransportError;
+use crate::io::emu::EmuState;
+use crate::io::uart::{Uart, UartError};
+use crate::transport::ti50emulator::Inner;
+use crate::transport::ti50emulator::Ti50Emulator;
+
+const TI50_UART_BAUDRATE: u32 = 115200;
+
+pub struct Ti50Uart {
+    inner: Rc<RefCell<Inner>>,
+    // This socket is valid as long as SubProcess is running.
+    socket: RefCell<Option<UnixStream>>,
+    // full path to socket file
+    path: PathBuf,
+    // last SubProcess ID
+    last_id: Cell<u64>,
+}
+
+impl Ti50Uart {
+    pub fn open(ti50: &Ti50Emulator, path: &str) -> Result<Self> {
+        let soc_path = ti50.inner.borrow().process.get_runtime_dir().join(path);
+        Ok(Self {
+            inner: ti50.inner.clone(),
+            socket: RefCell::new(None),
```

Comment:
FYI, this could also be `RefCell::default()` because RefCell will forward `::default` to the inner `T`.  In this case, that is `Option` which will default to None.
(I'm not advocating for a change here, just pointing out a different way this could be written)